frontend/src/metabase/public/containers/PublicOrEmbeddedDashboard/PublicOrEmbeddedDashboard.tsx

```diff
@@ -259,14 +273,30 @@ class PublicOrEmbeddedDashboardInner extends Component<PublicOrEmbeddedDashboard
           })}
           loading={!dashboard}
         >
-          {() =>
-            dashboard ? (
+          {() => {
+            if (!dashboard) {
+              return null;
+            }
+
+            if (!dashboardHasCards || !tabHasCards) {
+              return (
+                <DashboardEmptyStateWithoutAddPrompt
+                  isNightMode={isNightMode}
+                />
+              );
+            }
+
+            return (
               <DashboardContainer>
                 <DashboardGridConnected
                   dashboard={assoc(dashboard, "dashcards", visibleDashcards)}
                   isPublicOrEmbedded
-                  mode={PublicMode as unknown as Mode}
-                  selectedTabId={this.props.selectedTabId}
+                  mode={
+                    navigateToNewCardFromDashboard
+                      ? EmbeddingSdkMode
+                      : PublicMode
+                  }
+                  selectedTabId={selectedTabId}
```

Comment:
did we just simply forget to pass `selectedTabId` before and the static dashboard didn't work with multiple tab before? I want to understand why we pass this prop just now.

Reply:
Oh I see, I missed that entirely, sorry 🤦 . They were way too far apart on my screen.